public/app/plugins/datasource/cloudwatch/utils/query/getStatsGroups.ts

```diff
@@ -0,0 +1,38 @@
+const antlr4 = require('antlr4');
+const ScrollQLLexer = require('./ScrollQLLexer').ScrollQLLexer;
+const ScrollQLParser = require('./ScrollQLParser').ScrollQLParser;
+const ScrollQLParserListener = require('./ScrollQLParserListener').ScrollQLParserListener;
+
+class GroupListener extends ScrollQLParserListener {
+  groupNames: string[] = [];
+
+  enterLogStats(ctx: any) {
+    this.groupNames = [];
+    if (ctx.groups && ctx.groups.length > 0) {
+      const groups = ctx.groups;
+
+      groups.forEach((group: any) => {
+        const proj = group.fieldSpec?.().proj;
```

Comment:
Probably needs a bit of comments to explain with this does

Reply:
Sure, will add